src/compiler/scanner.ts

```diff
@@ -2464,986 +2497,1007 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
-
-            if (unicodeMode) {
-                // Annex B treats any unicode mode as the strict syntax.
-                annexB = false;
-            }
-
-            /** @see {scanClassSetExpression} */
-            let mayContainStrings = false;
-
-            /** The number of numeric (anonymous) capturing groups defined in the regex. */
-            let numberOfCapturingGroups = 0;
-            /** All named capturing groups defined in the regex. */
-            const groupSpecifiers = new Set<string>();
-            /** All references to named capturing groups in the regex. */
-            const groupNameReferences: (TextRange & { name: string; })[] = [];
-            /** All numeric backreferences within the regex. */
-            const decimalEscapes: (TextRange & { value: number; })[] = [];
-            /** A stack of scopes for named capturing groups. @see {scanGroupName} */
-            const namedCapturingGroups: Set<string>[] = [];
-
-            // Disjunction ::= Alternative ('|' Alternative)*
-            function scanDisjunction(isInGroup: boolean) {
-                while (true) {
-                    namedCapturingGroups.push(new Set());
-                    scanAlternative(isInGroup);
-                    namedCapturingGroups.pop();
-                    if (text.charCodeAt(pos) !== CharacterCodes.bar) {
-                        return;
-                    }
-                    pos++;
+    function scanRegularExpressionWorker(regExpFlags: RegularExpressionFlags, isUnterminated: boolean, annexB: boolean) {
+        // Why var? It avoids TDZ checks in the runtime which can be costly.
+        // See: https://github.com/microsoft/TypeScript/issues/52924
+        /* eslint-disable no-var */
+
+        /** Grammar parameter */
+        var unicodeSetsMode = !!(regExpFlags & RegularExpressionFlags.UnicodeSets);
+        /** Grammar parameter */
+        var unicodeMode = !!(regExpFlags & RegularExpressionFlags.UnicodeMode);
+
+        if (unicodeMode) {
+            // Annex B treats any unicode mode as the strict syntax.
+            annexB = false;
+        }
+
+        /** @see {scanClassSetExpression} */
+        var mayContainStrings = false;
+
+        /** The number of numeric (anonymous) capturing groups defined in the regex. */
+        var numberOfCapturingGroups = 0;
+        /** All named capturing groups defined in the regex. */
+        var groupSpecifiers: Set<string> | undefined;
+        /** All references to named capturing groups in the regex. */
+        var groupNameReferences: (TextRange & { name: string; })[] | undefined;
+        /** All numeric backreferences within the regex. */
+        var decimalEscapes: (TextRange & { value: number; })[] | undefined;
+        /** A stack of scopes for named capturing groups. @see {scanGroupName} */
+        var namedCapturingGroupsScopeStack: (Set<string> | undefined)[] = [];
+        var topNamedCapturingGroupsScope: Set<string> | undefined;
+

```

Comment:
Seems like this disable wasn't reenabled.

```suggestion
        /* eslint-enable no-var */
```